internal/db/remote/commit/templates/dump_initial_migration.sh

```diff
@@ -1,13 +1,17 @@
 #!/usr/bin/env bash
 set -euo pipefail
 
+# Explanation of special flags:
+#
+#   --schema-only     omit data like migration history, pgsodium key, etc.
+#   --exclude-schema  omit internal schemas as they are maintained by platform
+#   --no-comments     only object owner can set comment, omit to allow restore by non-superuser
+#   --extension '*'   prevents event triggers from being dumped, bash escaped with single quote
 pg_dump \
     --schema-only \
     --quote-all-identifier \
     --exclude-schema "$EXCLUDED_SCHEMAS" \
-    --schema '*' \
     --extension '*' \
+    --no-comments \
```

Comment:
We shouldn't omit *all* comments, only those on extensions. `postgres` can't comment on extensions (since they're created by `supabase_admin`), so we assume extension comments are unchanged.

Reply:
This is explained in comments of `dump_initial_migration.sh`. Only object owner can set comment. So any object owned by `supabase_admin` will need to be commented. Leaving it there is a time bomb as infra creates more migration scripts in the future.

Reply:
Sure, but that also removes all comments from users' objects. Anyway, aren't internal schemas supposed to be excluded in the first place?

Reply:
I think it's a trade off between getting complaints when cli users fail to `db push` vs complaints when user's object comments are not synced. The latter is less severe and would be my preference.

Perhaps we can add it back when we finally switch sql editor to postgres role.